### ISSUE TITLE
Publish next Zebra release: 1.4.0

### ISSUE THREAD

**Issue by mpguerra** [ZcashFoundation/zebra#7764]
# Prepare for the Release

These release steps can be done a week before the release, in separate PRs.
They can be skipped for urgent releases.

## State Full Sync Test

To check consensus correctness, we want to test that the state format is valid after a full sync. (Format upgrades are tested in CI on each PR.)

- [x] Make sure there has been [at least one successful full sync test](https://github.com/ZcashFoundation/zebra/actions/workflows/ci-integration-tests-gcp.yml?query=event%3Aschedule) since the last state change, or
- ~~Start a manual workflow run with a Zebra and `lightwalletd` full sync.~~

State format changes can be made in `zebra-state` or `zebra-chain`. The state format can be changed by data that is sent to the state, data created within the state using `zebra-chain`, or serialization formats in `zebra-state` or `zebra-chain`. 

After the test has been started, or if it has finished already:
- [x] Ask for a state code freeze in Slack. The freeze lasts until the release has been published.

## Checkpoints

For performance and security, we want to update the Zebra checkpoints in every release.
- [x] You can copy the latest checkpoints from CI by following [the zebra-checkpoints README](https://github.com/ZcashFoundation/zebra/blob/main/zebra-utils/README.md#zebra-checkpoints).

## Missed Dependency Updates

Sometimes `dependabot` misses some dependency updates, or we accidentally turned them off.

This step can be skipped if there is a large pending dependency upgrade. (For example, shared ECC crates.)

Here's how we make sure we got everything:
- [x] Run `cargo update` on the latest `main` branch, and keep the output
- [x] If needed, [add duplicate dependency exceptions to deny.toml](https://github.com/ZcashFoundation/zebra/blob/main/book/src/dev/continuous-integration.md#fixing-duplicate-dependencies-in-check-denytoml-bans)
- [ ] If needed, remove resolved duplicate dependencies from `deny.toml`
- [x] Open a separate PR with the changes
- [x] Add the output of `cargo update` to that PR as a comment

# Prepare and Publish the Release

Follow the steps in the [release checklist](https://github.com/ZcashFoundation/zebra/blob/main/.github/PULL_REQUEST_TEMPLATE/release-checklist.md) to prepare the release:

Release PR:
- [x] Update Changelog
    - [x] thank `@vuittont60` and `@rex4539` for their contributions
- [x] Update README
    - [x] Add to known issues: #7893
- [x] Update Zebra Versions
- [x] Update End of Support Height

Publish Release:
- [x] Create & Test GitHub Pre-Release
- [ ] Publish GitHub Release
- [ ] Publish Rust Crates
- [ ] Publish Docker Images

**Comment by teor2345** [comment]
> * Make sure there has been [at least one successful full sync test](https://github.com/ZcashFoundation/zebra/actions/workflows/continous-integration-docker.yml?query=event%3Aschedule) since the last state change, or

The last full sync was successful, but the lightwalletd full sync failed due to an unrelated devops bug #7796.

So I'm happy to call these done.

#7800 changes the sprout parameter data source we're using, so a full sync would be useful but not required. We've already checked they match the parameters used by `zcashd`.

**Comment by teor2345** [comment]
The last full sync failed due to the same bug:
https://github.com/ZcashFoundation/zebra/actions/runs/6666908268

**Comment by teor2345** [comment]
This is waiting on a manual full sync of Zebra and `lightwalletd`, because we've merged significant state and network changes since the last full sync:
https://github.com/ZcashFoundation/zebra/actions/runs/6686504232

**Comment by teor2345** [comment]
Here is the `cargo update` output:
```
$ cargo update
    Updating crates.io index             
    Updating ahash v0.8.3 -> v0.8.6      
    Updating async-trait v0.1.73 -> v0.1.74
    Updating base64 v0.21.4 -> v0.21.5
      Adding bindgen v0.68.1
    Updating clap v4.4.6 -> v4.4.7
    Updating clap_builder v4.4.6 -> v4.4.7
    Updating clap_derive v4.4.2 -> v4.4.7
    Updating clap_lex v0.5.1 -> v0.6.0
      Adding const-oid v0.9.5
    Updating cpufeatures v0.2.9 -> v0.2.11
    Updating cxx-gen v0.7.109 -> v0.7.110
      Adding der v0.7.8
    Updating deranged v0.3.8 -> v0.3.9
    Updating ed25519 v2.2.2 -> v2.2.3
    Updating fiat-crypto v0.2.1 -> v0.2.2
    Updating futures v0.3.28 -> v0.3.29
    Updating futures-channel v0.3.28 -> v0.3.29
    Updating futures-core v0.3.28 -> v0.3.29
    Updating futures-executor v0.3.28 -> v0.3.29
    Updating futures-io v0.3.28 -> v0.3.29
    Updating futures-macro v0.3.28 -> v0.3.29
    Updating futures-sink v0.3.28 -> v0.3.29
    Updating futures-task v0.3.28 -> v0.3.29
    Updating futures-util v0.3.28 -> v0.3.29
    Updating hashbrown v0.14.1 -> v0.14.2
    Updating hyper-rustls v0.24.1 -> v0.24.2
    Updating iana-time-zone v0.1.57 -> v0.1.58
    Updating ipnet v2.8.0 -> v2.9.0
    Updating mio v0.8.8 -> v0.8.9
    Updating parking_lot_core v0.9.8 -> v0.9.9
    Updating pest v2.7.4 -> v2.7.5
    Updating pest_derive v2.7.4 -> v2.7.5
    Updating pest_generator v2.7.4 -> v2.7.5
    Updating pest_meta v2.7.4 -> v2.7.5
      Adding pkcs8 v0.10.2
    Updating portable-atomic v1.4.3 -> v1.5.1
      Adding powerfmt v0.2.0
    Updating redox_syscall v0.3.5 -> v0.4.1
    Updating rgb v0.8.36 -> v0.8.37
      Adding ring v0.17.5
    Updating rustix v0.38.20 -> v0.38.21
    Updating rustls v0.21.7 -> v0.21.8
    Updating rustls-webpki v0.101.6 -> v0.101.7
    Updating sct v0.7.0 -> v0.7.1
    Updating serde v1.0.189 -> v1.0.190
    Updating serde_derive v1.0.189 -> v1.0.190
    Removing socket2 v0.4.9
    Removing socket2 v0.5.4
      Adding socket2 v0.4.10
      Adding socket2 v0.5.5
      Adding spki v0.7.2
    Updating tempfile v3.8.0 -> v3.8.1
    Updating time v0.3.29 -> v0.3.30
    Updating tokio-util v0.7.9 -> v0.7.10
    Updating toml v0.8.3 -> v0.8.6
    Updating toml_datetime v0.6.4 -> v0.6.5
    Updating toml_edit v0.20.3 -> v0.20.7
    Updating tracing-log v0.1.3 -> v0.1.4
      Adding untrusted v0.9.0
    Updating uuid v1.4.1 -> v1.5.0
    Removing windows v0.48.0
      Adding windows-core v0.51.1
    Updating winnow v0.5.16 -> v0.5.17
      Adding zerocopy v0.7.20
      Adding zerocopy-derive v0.7.20
```

**Comment by teor2345** [comment]
Oops must have been tagged on the wrong PR.

**Comment by teor2345** [comment]
The full sync passed in 2 days 2 hours:
https://github.com/ZcashFoundation/zebra/actions/runs/6686504232
